stat/roc.go

```diff
@@ -107,10 +107,10 @@ func ROC(cutoffs, y []float64, classes []bool, weights []float64) (tpr, fpr, thr
 	invNeg := 1 / nNeg
 	invPos := 1 / nPos
 	for i := range tpr {
-		tpr[i] *= invPos
-		tpr[i] = 1 - tpr[i]
-		fpr[i] *= invNeg
-		fpr[i] = 1 - fpr[i]
+		// Prevent fused float operations by
+		// making explicit float64 conversions.
+		tpr[i] = 1 - float64(tpr[i]*invPos)
```

Comment:
Is there a reason to prevent fused float operations other than making the results consistent between amd64 and arm64?

Reply:
Yeah, this is the thing that troubles me. The reason here is that using the FMA here results in a negative value. We know that ROC values must always be in [0, 1], so this is wrong. The question then is why the additional precision gives us this error and what we should do about it.

1. Either we're lucky with the examples that we have and the zeros are fortuitous or there is an interaction between the FMA and other parts the routine that give a discordance in the precisions and so these errors.
2. We make the code the same as it was and hope the second answer to 1 is correct or we round small negative values.

The approach I've taken seems simpler.

Reply:
*ping*

Reply:
I'll file an issue to look at this again. #1222